packages/context/src/resolver.ts

```diff
@@ -100,12 +101,15 @@ function resolve<T>(
         return injection.resolve(ctx, injection, s);
       } else {
         // Default to resolve the value from the context by binding key
-        return ctx.getValueOrPromise(injection.bindingKey, {
-          session: s,
-          // If the `optional` flag is set for the injection, the resolution
-          // will return `undefined` instead of throwing an error
-          optional: injection.metadata && injection.metadata.optional,
-        });
+        return ctx.getValueOrPromise(
```

Comment:
Let's keep the diff smaller.

```ts
const key = injection.bindingSelector as BindingAddress<unknown>;
return ctx.getValueOrPromise(key, {
  // ...
});
```

Same comment applies to other similar places in this file.